extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnChunkReaderImpl.java

```diff
@@ -315,7 +322,9 @@ private org.apache.parquet.format.Encoding getEncoding(final PageHeader pageHead
     }
 
     private PageHeader readPageHeader(final SeekableByteChannel ch) throws IOException {
-        try (final InputStream in = SeekableChannelsProvider.channelPositionInputStream(channelsProvider, ch)) {
+        // We don't know the exact size of page header, so we read it through a buffered stream in chunks of 128 bytes
+        try (final InputStream in = ChannelPositionInputStream.of(ch,
```

Comment:
I think this is a compelling argument to basically turn the int argument into a "hint", rather than a limit. Keep limits in the decompressor, let the channel provider make its own decision about how to apply hints, and make no guarantees about limits at that level.